collections/map.go

```diff
@@ -71,20 +71,20 @@ func (m Map[K, V]) Set(ctx context.Context, key K, value V) error {
 // errors with ErrNotFound if the key does not exist, or
 // with ErrEncoding if the key or value decoding fails.
 func (m Map[K, V]) Get(ctx context.Context, key K) (V, error) {
+	var v V
+
 	bytesKey, err := encodeKeyWithPrefix(m.prefix, m.kc, key)
 	if err != nil {
-		var v V
 		return v, err
 	}
 
 	kvStore := m.sa(ctx)
-	valueBytes := kvStore.Get(bytesKey)
-	if valueBytes == nil {
-		var v V
```

Comment:
the store does not error if the value does not exists, hence we need to retain the valueBytes == nil

Reply:
thanks